### ISSUE TITLE
Crashing on ipad

### ISSUE THREAD

**Issue by davequ** [merlos/iOS-Open-GPX-Tracker#168]
I can't get it to work on my ipad. It loads and shows the yellow splash screen, but then immediately exits. Tried re-installing but same issue. Ipad2 IOS 9.3.5.
Tried re-booting, then re-installing. Now it did run OK, so created a couple of waypoints, then when I edited one it crashed. And now wont start up again, as before

**Comment by vincentneo** [comment]
Hi @davequ,

I will take a look at this for the next couple of days. I should still have access to iOS 9.

**Comment by vincentneo** [comment]
Hi @davequ,

Indeed looks like a bug. Tested on iOS 9.

Related to #160, #152, #151, but this could be version specific, as it should be fixed on the latest commit, just not the App Store one, but am able to replicate with latest commit for this issue.

Issue appears to be:
```
Precondition failed: NSArray element failed to match the Swift Array Element type
Expected NSFetchRequestResult but found CDWaypoint: file /BuildRoot/Library/Caches/com.apple.xbs/Sources/swiftlang/swiftlang-1001.0.82.4/swift/stdlib/public/core/ArrayBuffer.swift, line 346
```
Meaning its another Core Data issue 🙃

**Comment by vincentneo** [comment]
UPDATE: looks like I found the possible fault of this entire issue: 
if you were to look at [NSFetchRequestResult](https://developer.apple.com/documentation/coredata/nsfetchrequestresult) API (which is the possible reason why converting from `NSFetchRequestResult` to an `NSManagedObject` failed), it officially lists iOS 10+ supported. I'm not too sure why it wasn't flagged out by the compiler when I wrote the code. Hopefully I would be able to find a workaround for this.
